src/hmac.cpp

```diff
@@ -0,0 +1,48 @@
+#include "hmac.hpp"
+#include <ios>
+// #include <openssl/hmac.h>
+// #include <openssl/evp.h>
+// #include <openssl/sha.h>
+#include "hmac_sha256.h"
+
+// This is a very slight hack I guess.
+extern "C" {
+#include "sha256.h"
+}
```

Comment:
Why is this necessary? Any downsides?

Reply:
That was a linker error that comes up due to differences in the results of symbol mangling. Either way it doesn't compile without it.

(I'll add a comment about this)